bots/acl_revert_modification.py

```diff
@@ -0,0 +1,189 @@
+    try:
+        network_acl.replace_entry(
+            CidrBlock=event_params.get('cidrBlock'),
+            DryRun=False,
+            Egress=event_params.get('egress'),
+            IcmpTypeCode=event_params.get('icmpTypeCode'),
+            PortRange=event_params.get('portRange'),
+            Protocol=event_params.get('aclProtocol'),
+            RuleAction=event_params.get('ruleAction'),
+            RuleNumber=event_params.get('ruleNumber'))
+
+        text_output = f"Rule number {params.get('ruleNumber')} successfully replaced \n"
+
+    except ClientError as e:
+        text_output = f'Unexpected error: {e}\n'
+
+
+    return text_output
+
+
+def create_entry(boto_session, params):
+    '''
+    function will be called if event is: DeleteNetworkAclEntry.
+    it will recreate the deleted entry.
+    To get all the params to create a new entry function will search for the creation of the deleted entry.
+
+    In this function params are not used in function create_entry because they lack information needed to create an entry.
+    '''
+
+    text_output = ''
+    acl_id = params.get('networkAclId')
+    # Trying to get entry creation event
+    event_name, event_params = find_event(boto_session, acl_id)
+
+    # Exiting if creation event can not be found
+    if event_name != 'CreateNetworkAclEntry':
+        text_output = f'Can not recreate deleted rule {params.get("ruleNumber")}.\nExiting'
+        return text_output
+
+    ec2_resource = boto_session.resource('ec2')
+    network_acl = ec2_resource.NetworkAcl(acl_id)
+
+    try:
+        network_acl.create_entry(
+            CidrBlock=event_params.get('cidrBlock'),
+            DryRun=False,
+            Egress=event_params.get('egress'),
+            IcmpTypeCode=event_params.get('icmpTypeCode'),
+            PortRange=event_params.get('portRange'),
+            Protocol=event_params.get('aclProtocol'),
+            RuleAction=event_params.get('ruleAction'),
+            RuleNumber=event_params.get('ruleNumber'))
+
+        text_output = f"Rule number {params.get('ruleNumber')} successfully created \n"
+
+    except ClientError as e:
+
+        text_output = f'Unexpected error: {e}\n'
+
+
+    return text_output
+
+
+def delete_entry(boto_session, params):
+    '''
+    function will be called if event is: CreateNetworkAclEntry
+    it will delete the new created entry.
+    '''
+
+    text_output = ''
+    ec2_resource = boto_session.resource('ec2')
+    network_acl = ec2_resource.NetworkAcl(params.get('networkAclId'))
+
+    try:
+        network_acl.delete_entry(
+            DryRun=False,
+            Egress=params.get('egress'),
+            RuleNumber=params.get('ruleNumber'))
+        text_output = f"Rule number {params.get('ruleNumber')} successfully deleted \n"
+
+    except ClientError as e:
+
+        text_output = f'Unexpected error: {e}\n'
+
+
+    return text_output
+
+
+def find_event(boto_session, attribute_value, entity={}):

```

Comment:
Should be as util - and why there are prints in here?

Reply:
I need to use this function in two different situations. one time I use the utils func and other times when I dont have a time, I use look up and check all events related to the acl.

Reply:
@doronzil What do you think about this function ?

Reply:
You can change "cloudtrail_event_lookup" function in utils to support "no time" lookups. please edit it and use it :)